src/mp4box.js

```diff
@@ -183,7 +184,10 @@ MP4Box.prototype.processSamples = function() {
 					Log.info("MP4Box", "Sending fragmented data on track #"+fragTrak.id+" for samples ["+Math.max(0,trak.nextSample-fragTrak.nb_samples)+","+(trak.nextSample-1)+"]"); 
 					Log.info("MP4Box", "Sample data size in memory: "+this.inputIsoFile.getAllocatedSampleDataSize()); 			
 					if (this.onSegment) {
-						this.onSegment(fragTrak.id, fragTrak.user, fragTrak.segmentStream.buffer, trak.nextSample);
+						/* get the time for this segment */
+						sample = trak.nextSample < trak.samples.length && trak.samples[trak.nextSample];
+						var time = (sample.dts + sample.duration) / sample.timescale;
```

Comment:
at this stage in the processing, nextSample is the first sample of the next fragment that will be created. I'm not sure that's what you want. Also you may want the composition time and not the decode time.